Fix a typo (#29135)
dotnet/docs: docs/csharp/whats-new/csharp-9.md

C# 9 includes new pattern matching improvements:

> - ***Type patterns*** match a variable that is a particular type

Reviewer: I think this could be misleading. The variable already has a type. The ***Type pattern*** checks that the object a variable refers to matches a type. How about this:

```suggestion
- ***Type patterns*** match an object matches a particular type
```
